### PR TITLE
GH-12702: Add new command response parameter: "skip_slack_parsing".

### PR DESCRIPTION
**Pull request by somenet**
#### Summary
If command-response contains ``"skip_slack_parsing":"true"`` the command-response hander shall skip the slack-compatibility parsing

#### Ticket Link
- Fixes https://github.com/mattermost/mattermost-server/issues/12702
- https://mattermost.atlassian.net/browse/MM-19540

### CODE REVIEW

**Comment by somenet** [comment]
manually tested with: https://github.com/somenet/mattermost-dev-env

**Comment by somenet** [comment]
Proposed docs change: https://github.com/mattermost/mattermost-developer-documentation/pull/456

File: model/command_response.go [hunk 1]
```diff
@@ -18,17 +18,18 @@ const (
...
+	ResponseType     string             `json:"response_type"`
+	Text             string             `json:"text"`
+	Username         string             `json:"username"`
+	ChannelId        string             `json:"channel_id"`
+	IconURL          string             `json:"icon_url"`
+	Type             string             `json:"type"`
+	Props            StringInterface    `json:"props"`
+	GotoLocation     string             `json:"goto_location"`
+	TriggerId        string             `json:"trigger_id"`
+	SkipSlackParsing string             `json:"skip_slack_parsing"`
```
Comment (hanzei) [inline model/command_response.go]: Why is this a `string`? The allowed values are `true` and `false` right? What do you think about making it a `bool` instead?

File: app/command.go [hunk 2]
```diff
@@ -435,8 +435,8 @@ func (a *App) HandleCommandResponsePost(command *model.Command, args *model.Comm
 		post.AddProp("from_webhook", "true")
 	}
 
-	// Do not process text if this is a code block
-	skipSlackParsing := command.Trigger == "code"
+	// Do not process text if this is a code block or response contains "skip_slack_parsing":"true"
+	skipSlackParsing := command.Trigger == "code" || (len(response.SkipSlackParsing) != 0 && response.SkipSlackParsing == "true")
```
Comment (hanzei) [inline app/command.go]: Would you mind removing the special rule for `code` and update `app/command_code.go` to use `SkipSlackParsing`?

**Comment by hanzei** [comment]
@lieut-data Given that this is a new feature I would opt to not include this into 5.19. Are you fine with this?

File: app/command.go (hunk 2 above)
Comment (somenet) [reply app/command.go]: done.

File: model/command_response.go (hunk 1 above)
Comment (somenet) [reply model/command_response.go]: done.

**Comment by somenet** [comment]
This is intended to be a bug-fix of wrong ``/command``-response handling.

I dont particularly care if/how fast it makes it into official mattermost, but it would be cool, if someone compiled it with E10-features for me and got me the bin. :)

We need this for advent of code and other student activities, where we send code snippets around via bot commands.

**Comment by somenet** [comment]
Also: how do i fix: ``mattermost/serverless-ops/github — This PR needs to sync and merge``?

**Comment by somenet** [comment]
ok... not like this... Lets kill this one, I guess...

**Comment by somenet** [comment]
Second attempt: https://github.com/mattermost/mattermost-server/pull/13420

File: app/command.go (hunk 2 above)
Comment (somenet) [reply app/command.go]: after botching this PR, i created a new one.

Is there a useful use-case that not all built in commands should be excluded from the Slack-Compatibility layer?

**Pull request closed without merging**